### ISSUE TITLE
Feature request: rename target from link

### ISSUE THREAD

**Issue by pacien** [artempyanykh/marksman#154]
Currently, it is possible to use the rename action on a title to have all the
links referencing that file updated.

However, using the rename action on the link in another file does not seem to
do anything (neither changing the link nor the title of the target).

This would be very convenient to be able to rename files from some index or
reference page.

**Comment by artempyanykh** [comment]
I agree. I think the part that needs to be thought through a bit better is the target of the rename. For instance,
```
file: Hello World.md
---

# Hello World

...
```

```
file: Goodbye World.md
---
# Goodbye World

This is a follow up to [[Hello World]].
                              ^
                              |-- rename this
```

This wiki link could be resolved as a file name link or as a title link. When you hit rename what should be renamed: the title of 'Hello World.md' document or the name of this document? Potentially, we could use the completion style to disambiguate here.

Anyhow, I agree this would be convenient; it just needs some careful implementation.
